src/main/scala/com/amazon/deequ/analyzers/Analyzer.scala

```diff
@@ -260,7 +260,7 @@ case class AnalyzerOptions(nullBehavior: NullBehavior = NullBehavior.Ignore)
 
 object NullBehavior extends Enumeration {
   type NullBehavior = Value
-  val Ignore, EmptyString, Fail = Value
+  val Ignore, EmptyString, Fail, NegVal = Value
```

Comment:
It would probably be better to separate this into two enums, one for strings and a different for columns. That's because `EmptyString` makes no sense for numbers and `NegVal` makes no sense for strings. As it stands now, if I pass `NullBehavior.EmptyString` to the `Maximum` analyzer, I get no warning and unexpected behavior.

Another option would be to implement a method inside the enum that is given a Spark type and returns the sentinel value for that type (if one is defined) or throws. For example, for [numeric types](https://spark.apache.org/docs/3.2.2/sql-ref-datatypes.html), it returns `-1`. For String type, it returns `""`. For anything else, it throws. I am not entirely sure that would work though, as I don't know if Analyzers know the type they're working on

There may also be a way to let users override the default sentinel but may not be worth implementing in this PR if it's too complicated

Reply:
On another note, I think `Ignore` and `Fail` are misleading as names (I know you didn't create this enum but as I look at it this occurs to me).

What we really mean is, `Ignore` will produce a result that doesn't take into account `null` values, and `Fail` will preserve `null` values as they are and that may fail down the line. May be worth renaming `Fail` to `Preserve`, as an opposite to "Replace" in this context